client/src/main/scala/org/apache/celeborn/client/ChangePartitionManager.scala

```diff
@@ -278,14 +282,42 @@ class ChangePartitionManager(
       }
     }
 
-    // Get candidate worker that not in excluded worker list of shuffleId
-    val candidates =
-      lifecycleManager
-        .workerSnapshots(shuffleId)
-        .keySet()
-        .asScala
-        .filter(lifecycleManager.workerStatusTracker.workerAvailable)
-        .toList
+    var candidates = new util.HashSet[WorkerInfo]()
+    if (changPartitionWithAvailableWorkers) {
```

Comment:
candidates should also .filter(lifecycleManager.workerStatusTracker.workerAvailable